keras/preprocessing/image.py

```diff
@@ -821,6 +823,71 @@ def next(self):
         return batch_x, batch_y
 
 
+def _count_valid_files_in_directory(dirpath, white_list_formats, follow_links):
+    """Count files with extension in `white_list_formats` contained in a directory.
+
+    # Arguments
+        dirpath: absolute path to the directory containing files to be counted
+        white_list_formats: set of strings containing allowed extensions for
+            the files to be counted.
+
+    # Returns
+        the count of files with extension in `white_list_formats` contained in
+        the directory.
+    """
+    def _recursive_list(subpath):
+        return sorted(os.walk(subpath, followlinks=follow_links), key=lambda tpl: tpl[0])
+
+    samples = 0
+    for root, _, files in _recursive_list(dirpath):
+        for fname in files:
+            is_valid = False
+            for extension in white_list_formats:
+                if fname.lower().endswith('.' + extension):
+                    is_valid = True
+                    break
+            if is_valid:
+                samples += 1
+    return samples
+
+
+def _list_valid_filenames_in_directory(subdir, directory, white_list_formats,
```

Comment:
Which function are you referring to?
If it's _list_valid_[...] then I passed both subdir and directory to make it more explicit that the filenames in the returned list are relative to directory.
My original version took just the absolute path to subdir, but I think this is clearer and simpler to document (although possibly prone to errors if subdir is not actually contained in directory).
What do you think?